core/modules/wdb/QueryRunner.cc

```diff
@@ -110,6 +110,9 @@ QueryRunner::QueryRunner(wbase::Task::Ptr const& task,
 bool QueryRunner::_initConnection() {
     mysql::MySqlConfig localMySqlConfig(_mySqlConfig);
     localMySqlConfig.username = _task->user; // Override with czar-passed username.
+    if (_mysqlConn != nullptr) {
+        LOGS(_log, LOG_LVL_ERROR, "QueryRunner::_initConnection _mysqlConn not nullptr _mysqlConn=" << _mysqlConn.get());
+    }
```

Comment:
Is this really possible, and if so then why? If this is happening because of a logic error in the implementation then the process must be aborted right away. If this is happening because of a memory corruption - the same thing, abort right away to investigate the problem rather than hiding it "under the carpet". If this is because of race condition - then a reason for the race condition had to be investigated.  In aby case, aborting the application would be the only right step after discovering and reporting this condition.

Reply:
It isn't supposed to be possible, but it, or something similar happened in one of the cluster crashes. The following line called the destructor, which shouldn't happen. So this is looking for some insight into what went wrong if it happens again.